[doc][2.25.1] PG15 manual upgrade (#26390)
yugabyte/yugabyte-db: docs/content/preview/manage/upgrade-pg15.md

---
title: Upgrade YugabyteDB
headerTitle: Upgrade YugabyteDB to PostgreSQL 15
linkTitle: Upgrade to PG15
description: Upgrade YugabyteDB to PostgreSQL 15
headcontent: Upgrade YugabyteDB to a version that supports PG 15
menu:
  preview:
    identifier: upgrade-pg15
    parent: manage-upgrade-deployment
    weight: 706
type: docs
---

Upgrading YugabyteDB from a version based on PostgreSQL 11 (all versions prior to v2.25) to a version based on PostgreSQL 15 (currently v2.25.1) requires some additional steps.

## Before you begin

{{< warning >}}
v2.25.1.0 is a preview release that is only meant for evaluation purposes and should not be used in production.
{{< /warning >}}

> - Do not execute DDL statements during the upgrade.

Reviewer: ```suggestion
- DDL statements are blocked for the duration of the upgrade.
```

[Roll back all YB-Tservers](../upgrade-deployment/#1-roll-back-yb-tservers) to v2024.2.2.0.

### Roll back the catalog

Use the v2.25.1.0 yb-admin command `rollback_ysql_major_version_upgrade` to roll back the YSQL catalog. Use the command as follows:

```sh
yb-admin --master_addresses <master_addresses> rollback_ysql_major_version_upgrade
```

For example:

```sh
./yugabyte-2.25.1.0/bin/yb-admin --master_addresses 127.0.0.1:7100,127.0.0.2:7100,127.0.0.3:7100 rollback_ysql_major_version_upgrade
```

```output
Rollback successful
```

### Roll back YB-Masters to v2024.2.2.0

[Roll back all YB-Masters](../upgrade-deployment/#2-roll-back-yb-masters) to v2024.2.2.0.

## Post upgrade

After the upgrade is done, you can re-enable expression pushdowns. For example:

```sh
./bin/yb-ts-cli set_flag --server_address 127.0.0.1:9100 --force ysql_yb_enable_expression_pushdown true

./bin/yb-ts-cli set_flag --server_address 127.0.0.2:9100 --force ysql_yb_enable_expression_pushdown true

./bin/yb-ts-cli set_flag --server_address 127.0.0.3:9100 --force ysql_yb_enable_expression_pushdown true
```

Run ANALYZE on your databases to get the most optimal query plans.

## Limitations

> - No yugabyted and Yugabyte Anywhere support. {{<issue 24793>}}

Reviewer: yugabyted is not available. Can we update this doc to add the.
The partial steps are in https://yugabyte.slack.com/files/URNU51UH1/F08G7403RFB/pg15_upgrade_oss.sh

https://yugabyte.slack.com/archives/C07DG9B7C3T/p1741021856125019

Reviewer: Lets not mention YBA. This is the OSS doc

- YB-Tservers need to run on all YB-Master nodes. Kubernetes and Docker are not yet supported. {{<issue 24719>}}

> - DDL blocking is not enforced by the DB. {{<issue 25362>}}

Reviewer: These are fixed

- Temporary tables, refreshing materialized views, and expression pushdown are not available. {{<issue 24731>}}, {{<issue 24732>}}, {{<issue 24730>}}

> - Upgrading with extensions is not supported. {{<issue 24733>}}

Reviewer: ```suggestion
- Upgrading with extensions is not yet supported. {{<issue 24733>}}
```

Reviewer: @fizaaluthra is there a subset of extensions that are safe even without your change?

> - Analyze must be run after the upgrade. {{<issue 25721>}}

Reviewer: Only needed if customer is using CBO
